### PR TITLE
Modify our Javascript to enable bundling

### PR DESCRIPTION
**Pull request by greg-does-weather**
## What does this PR do? 🛠️

This PR changes our Javascript to use ESM modules. It creates a single entrypoint script for each type of page (front page, AFD, and point) which then imports the necessary scripts. The primary theme's library list is updated to remove all the individual components and instead reference the page entrypoints. The page templates are updated to attach only their page entrypoints.

In development, all of the other scripts are still served independently, and there is no build step in order to see changes. The browser continues to load all of the component scripts individually as they are `import`ed from the entrypoint.

In sandbox deployments, the entrypoints are bundled, replacing the original entrypoint file, and delivered to cloud.gov. The practical result is that each page only needs to load a single Javascript file in order to import all of our custom functionality because the library list points at the entrypoint file which is now a bundle of all of the component files required by that page.

In CI/CD testing, the entrypoints are determined dynamically from the library list and are build into single bundled, minified scripts. Tests are then run against the monolithic Javascript. I opted to make this change for testing since I would like to eventually see this bundling and minification pushed to staging and beta, and it'd be good to test against what we're actually deploying.

### CODE REVIEW

File: web/themes/new_weather_theme/assets/js/charts/hourly-pops.js [hunk 1]
```diff
@@ -1,98 +1,86 @@
 /* global Chart ChartDataLabels */
-(() => {
```
Comment (jamestranovich-noaa) [inline web/themes/new_weather_theme/assets/js/charts/hourly-pops.js:2]: Looks like we have some charting stuff that was accidentally included here

File: docs/dev/javascript-bundling.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+We use ESM modules for our custom code. This helps create clean scope boundaries
+so we don't have to worry about polluting the global scope. It also helps us
+manage import order, since esbuild can just figure it out for us.
+
+## Drupal libraries
+
+We define what Javascript to deploy in the Drupal theme's libraries file at
+`web/themes/new_weather_theme/new_weather_theme.libraries.yml`. For custom code,
+there should be one entry per page type. The same code may be bundled in multiple
+page types, but we only want to deliver a single bundled file per page type.
```
Comment (jamestranovich-noaa) [inline docs/dev/javascript-bundling.md:18]: Can we add something about the fact that we need the yml key to end in `-page` to actually bundle?

**Review by jamestranovich-noaa — CHANGES_REQUESTED** [review]

File: web/themes/new_weather_theme/assets/js/charts/hourly-pops.js (hunk 1 above)
Comment (mgwalker) [reply web/themes/new_weather_theme/assets/js/charts/hourly-pops.js:2]: Not an accident! These were merged into main a little bit ago, so I needed to tweak them for the bundling to work correctly with them.

File: docs/dev/javascript-bundling.md (hunk 2 above)
Comment (greg-does-weather) [reply docs/dev/javascript-bundling.md:18]: Yes, thank you! I was struggling to remember what needed to be documented. Slow start with the Monday brain and a gazillion meetings. 😬

**Review by jamestranovich-noaa — APPROVED** [review]
LGTM! excited to see this work land!

**Review by eric-gade — APPROVED** [review]
I hear ESM is the future.

LGTM 👍 

## Code Review Checklist

This is an automated comment on every pull request requiring a review. A checked item is either an assertion that I have tested this item or an indication that I have verified it does not apply to this pull request.

## The Basics
- [x] Checks are passing
- [x] I read the code
- [x] I ran the code
- [x] (if applicable) Post deploy steps are run
- [x] (if applicable) I validated the change on the deployed version in <environment>

## Documentation
- [x] changes to “how we do things” are documented in READMEs
- [x] all new functions and methods are commented using plain language
- [x] any new modules added documented in modules.md

## Security
- [x] security false positives are documented
- [x] data from external sources is cleaned and clearly marked

## Reliability
- [x] error handling exists for unusual or missing values
- [x] interactions with external systems are wrapped in try/except
- [x] functionality is tested with unit or integration tests
- [x] dependency updates in composer.json also got changed in composer-lock.json

## Infrastructure
- [x] all changes are auditable and documented via a script
- [x] it is clear who can and should run the script
- [x] (if applicable) diagrams have been updated or added in PlantUML

## Accessibility
- [x] New pages have been added to cypress-axe file so that they will be tested with our automated accessibility testing
- [x] Meets WCAG 2.0 AA or 2.1 AA for Section 508 compliance
    - [ ] Site is keyboard accessible. All interactions can be accessed with a keyboard
    - [ ] Site is free of keyboard traps. The keyboard focus is never trapped in a loop
    - [ ] All form inputs have explicit labels
    - [ ] Form instructions are associated with inputs
    - [ ] All relevant images use an img tag
    - [ ] All images have appropriate alt attributes
    - [ ] Multimedia is tagged. All multimedia has appropriate captioning and audio description
    - [ ] Text has sufficient color contrast. All text has a contrast ratio of 4.5:1 with the background
    - [ ] Site never loses focus. Focus is always visible when moving through the page with the keyboard
    - [ ] Tab order is logical
    - [ ] Tables are coded properly. Tables have proper headers and column attributes
    - [ ] Headings are nested properly. Heading elements are nested in a logical way
    - [ ] Language is set. The language for the page is set
    - [ ] CSS is not required to use the page. The page makes sense with or without CSS
    - [ ] Links are unique and contextual. All links can be understood taken alone, e.g., ‘Read more - about 508’
    - [ ] Page titles are descriptive

## Device Matrix
- [x] firefox/gecko (renders correctly and user interactions work)
- [x] chrome/chromium/edge (renders correctly and user interactions work)
- [x] safari/webkit (renders correctly and user interactions work)
- [x] web page is readable and usable 
    - [ ] at 480px     (mobile)
    - [ ] at 640px     (tablet)
    - [ ] at 1024px    (desktop)

**Pull request merged**